tensorflow/python/ops/ctc_ops.py

```diff
@@ -47,6 +47,18 @@ def ctc_loss(inputs, labels, sequence_length,
   max(labels.indices(labels.indices[:, 1] == b, 2))
     <= sequence_length(b) for all b.
   ```
+  
+  Useful Tips:
+  
+  The ctc_loss has as input an affine projection of RNN state:
+    inputs_t = W*h^L_{t} + b, for t = {1,...,max_time}
+  where h is the hidden state of last rnn calculated at timestep t.
+  
+  The `inputs` argument has size num_labels + 1 where +1 is reserved to
+  blank label, e.g., for a vocab = {a, b, c} the num_classes will be 4 
+  where {a: 0, b: 1, c: 2, blank: 3}.
+  
+  `ctc_loss` function already interleaves blank label between the target labels.
```

Comment:
I think this is an important note, he points out that you *don't* need to encode the target labels (by interleaving with blank symbol) before passing to ctc_loss, this is internally done for you!

it took me a while to grasp this from the Cpp code, since there is no examples for ctc loss/decode currently (you can only find the straight forward usage in the test code)